workbench-core/datasets/src/dataSetsAuthorizationPlugin.ts

```diff
@@ -55,8 +55,11 @@ export interface DataSetsAuthorizationPlugin {
   /**
    * Remove all permssions from a given dataset.
    * @param datasetId - the ID of the datset for which permissions are to be removed.
+   * @param authenticatedUser - an object representing the user making the request.
    * @returns a {@link PermissionsResponse} indicating the permissions removed.
-   * IMPORTANT: If the returned object has a non-null `pageToken`, there are more permissions to delete.
    */
-  removeAllAccessPermissions(datasetId: string): Promise<PermissionsResponse>;
+  removeAllAccessPermissions(
+    datasetId: string,
+    authenticatedUser: { id: string; roles: string[] }
```

Comment:
I'm seeing this definition in multiple places, might be worth making a type for it.

Reply:
Yes! At some point, we should refactor our reference implementation plugins into separate packages so our packages are not bound to each other. Ideally that change should affect only the authz plugin, and not the interface. If I pull `AuthenticatedUser` from `workbench-core-authorization` now, I will have to make the change to the interface later to tease the packages apart. I favor not setting up future interface changes even if they seem trivial.

Reply:
It will be naturally coupled by the way we depend on authenticatedUser but I agree